core/dbt/contracts/graph/manifest.py

```diff
@@ -439,11 +441,34 @@ def patch_nodes(self, patches):
         # nodes looking for matching names. We could use _find_by_name if we
         # were ok with doing an O(n*m) search (one nodes scan per patch)
         for node in self.nodes.values():
-            if node.resource_type != NodeType.Model:
+            if node.resource_type == NodeType.Source:
                 continue
             patch = patches.pop(node.name, None)
             if not patch:
                 continue
+            expected_key = node.resource_type.pluralize()
+            if expected_key == patch.yaml_key:
+                node.patch(patch)
+            if expected_key != patch.yaml_key:
+                if patch.yaml_key == 'models':
+                    deprecations.warn(
+                        'models-key-mismatch',
+                        patch=patch, node=node, expected_key=expected_key
+                    )
+                else:
+                    msg = printer.line_wrap_message(
+                        f'''\
+                        '{node.name}' is a {node.resource_type} node, but it is
```

Comment:
This renders out in a pretty funky way:

<img width="896" alt="Screen Shot 2020-01-20 at 9 58 24 PM" src="https://user-images.githubusercontent.com/1541139/72771939-7e836c80-3bd0-11ea-8476-3eb6334282c8.png">

^ i actually had my printer width configured to be 90 chars, but when i delete that config, it still looks like:
<img width="838" alt="Screen Shot 2020-01-20 at 10 05 42 PM" src="https://user-images.githubusercontent.com/1541139/72772076-08333a00-3bd1-11ea-88e6-9581a95d8a8c.png">

I didn't dig so deep into this, but it looks like the issue is at least partially caused by the `* Deprecation Warning` part of the string, which is _not_ indented.

Once this is fixed up, the PR LGTM!